Source/Core/Core/IOS/ES/TitleInformation.cpp

```diff
@@ -205,6 +205,38 @@ IPCCommandResult ES::GetBoot2Version(const IOCtlVRequest& request)
   Memory::Write_U32(4, request.io_vectors[0].address);
   return GetDefaultReply(IPC_SUCCESS);
 }
+
+IPCCommandResult ES::GetSharedContentsCount(const IOCtlVRequest& request) const
+{
+  if (!request.HasNumberOfValidVectors(0, 1) || request.io_vectors[0].size != sizeof(u32))
+    return GetDefaultReply(ES_EINVAL);
+
+  const u32 count = IOS::ES::GetSharedContentsCount();
+  Memory::Write_U32(count, request.io_vectors[0].address);
+
+  INFO_LOG(IOS_ES, "GetSharedContentsCount: %u contents", count);
+  return GetDefaultReply(IPC_SUCCESS);
+}
+
+IPCCommandResult ES::GetSharedContents(const IOCtlVRequest& request) const
+{
+  if (!request.HasNumberOfValidVectors(1, 1) || request.in_vectors[0].size != sizeof(u32))
+    return GetDefaultReply(ES_EINVAL);
+
+  const u32 max_count = Memory::Read_U32(request.in_vectors[0].address);
+  if (request.io_vectors[0].size != 20 * max_count)
+    return GetDefaultReply(ES_EINVAL);
+
+  const std::vector<std::array<u8, 20>> hashes = IOS::ES::GetSharedContents();
+  const u32 count = std::min(static_cast<u32>(hashes.size()), max_count);
+  for (u32 i = 0; i < count; ++i)
```

Comment:
Nice catch. I think I had debug logging at one point in the loop and forgot to simplify the whole thing after removing it.